tests/test_app.py

```diff
@@ -138,42 +106,70 @@ def remove_dir(self, path):
 def test_path_cases(username_first, username_rest):
     username = username_first + username_rest
     assert (
-            username + "/foo/bar" == utils.Path.validate_path(username, "foo/bar").user_path
+        username + "/foo/bar"
```

Comment:
at some point these should be parametrised... not this PR, though, it is big enough already! 😄